### ISSUE TITLE
Cannot `import tensorflow.summary` since 2019-06-08 nightly

### ISSUE THREAD

**Issue by wchargin** [tensorflow/tensorflow#29608]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): gLinux
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): `('v1.12.1-3679-g3040de1372', '2.0.0-dev20190608')`
- Python version: 2.7 or 3.6
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**

Importing `tensorflow.summary` raises `ModuleNotFoundError`:

```
$ python -c 'import tensorflow.summary'; echo $?
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'tensorflow.summary'
```

**Describe the expected behavior**

Importing that module should succeed:

```
$ python -c 'import tensorflow.summary'; echo $?
0
```

**Code to reproduce the issue**

```
$ activate ~/virtualenv/tf-nightly-2.0-preview-20190606-py2.7
$ python -c 'import tensorflow.summary'; echo $?
0
$ activate ~/virtualenv/tf-nightly-2.0-preview-20190607-py2.7
$ python -c 'import tensorflow.summary'; echo $?
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/buildtools/current/sitecustomize/sitecustomize.py", line 152, in SetupPathsAndImport
    return real_import(name, globals, locals, fromlist, level)
  File "/HOMEDIR/virtualenv/tf-nightly-2.0-preview-20190607-py2.7/local/lib/python2.7/site-packages/tensorflow/__init__.py", line 93, in <module>
    from tensorflow_core import *
AttributeError: 'module' object has no attribute 'compiler'
1
$ activate ~/virtualenv/tf-nightly-2.0-preview-20190608-py2.7
$ python -c 'import tensorflow.summary'; echo $?
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/buildtools/current/sitecustomize/sitecustomize.py", line 152, in SetupPathsAndImport
    return real_import(name, globals, locals, fromlist, level)
ImportError: No module named summary
1
```

**Other info / logs**

cc @mihaimaruseac; is this related to recent Pip changes?

**Comment by mihaimaruseac** [comment]
I think it is, sorry about that. Will get to look into this in around 1 hour

**Comment by wchargin** [comment]
Cool, nothing pressing; we’ll just stay pinned to 2019-06-06 for now.

**Comment by nfelt** [comment]
@mihaimaruseac the `tf.summary` import mechanism is pretty arcane - let me know if I can be helpful with any debugging/fixing here.

**Comment by mihaimaruseac** [comment]
It seems I cannot reproduce with a simple virtualenv, maybe I'm missing something

```bash
mihaimaruseac@ankh:/tmp/virtual_pip$ mkdir tf2.0
mihaimaruseac@ankh:/tmp/virtual_pip$ cd tf2.0
mihaimaruseac@ankh:/tmp/virtual_pip/tf2.0$ virtualenv .
Running virtualenv with interpreter /usr/bin/python2
New python executable in /tmp/virtual_pip/tf2.0/bin/python2
Also creating executable in /tmp/virtual_pip/tf2.0/bin/python
Installing setuptools, pkg_resources, pip, wheel...sdone.
omihaimaruseac@ankh:/tmp/virtual_pip/tf2.0$ source bin/activate
(tf2.0) mihaimaruseac@ankh:/tmp/virtual_pip/tf2.0$ pip install tf-nightly-2.0-preview
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Collecting tf-nightly-2.0-preview
  Using cached https://files.pythonhosted.org/packages/5c/aa/92b2a8d290cffe1f195b055ffb5ea66e99cec2c532cc867f440baa9328e0/tf_nightly_2.0_preview-2.0.0.dev20190608-cp27-cp27mu-manylinux1_x86_64.whl
Collecting tb-nightly<1.15.0a0,>=1.14.0a0 (from tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/8a/44/6c99eee8359af17ae32deeabb4ab3660e2bf1fa1f10f9947225915a00eec/tb_nightly-1.14.0a20190610-py2-none-any.whl
Collecting backports.weakref>=1.0rc1 (from tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/88/ec/f598b633c3d5ffe267aaada57d961c94fdfa183c5c3ebda2b6d151943db6/backports.weakref-1.0.post1-py2.py3-none-any.whl
Collecting mock>=2.0.0 (from tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/05/d2/f94e68be6b17f46d2c353564da56e6fb89ef09faeeff3313a046cb810ca9/mock-3.0.5-py2.py3-none-any.whl
Requirement already satisfied: wheel in ./lib/python2.7/site-packages (from tf-nightly-2.0-preview) (0.33.4)
Collecting absl-py>=0.7.0 (from tf-nightly-2.0-preview)
Collecting numpy<2.0,>=1.14.5 (from tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/1f/c7/198496417c9c2f6226616cff7dedf2115a4f4d0276613bab842ec8ac1e23/numpy-1.16.4-cp27-cp27mu-manylinux1_x86_64.whl
Collecting termcolor>=1.1.0 (from tf-nightly-2.0-preview)
Collecting protobuf>=3.6.1 (from tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/b2/a8/ad407cd2a56a052d92f602e164a9e16bede22079252af0db3838f375b6a8/protobuf-3.8.0-cp27-cp27mu-manylinux1_x86_64.whl
Collecting google-pasta>=0.1.6 (from tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/35/95/d41cd87d147742ef72d5d1dc317318486e3fbffdadf24a60e70dedf01d56/google_pasta-0.1.7-py2-none-any.whl
Collecting gast>=0.2.0 (from tf-nightly-2.0-preview)
Collecting enum34>=1.1.6 (from tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/c5/db/e56e6b4bbac7c4a06de1c50de6fe1ef3810018ae11732a50f15f62c7d050/enum34-1.1.6-py2-none-any.whl
Collecting grpcio>=1.8.6 (from tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/8e/62/448199c50e05526f7fd336d8a87bd83a0c7d92188ae51ebea23aa5f3c37e/grpcio-1.21.1-cp27-cp27mu-manylinux1_x86_64.whl
Collecting six>=1.10.0 (from tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting keras-applications>=1.0.6 (from tf-nightly-2.0-preview)
Collecting wrapt>=1.11.1 (from tf-nightly-2.0-preview)
Collecting astor>=0.6.0 (from tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/d1/4f/950dfae467b384fc96bc6469de25d832534f6b4441033c39f914efd13418/astor-0.8.0-py2.py3-none-any.whl
Collecting keras-preprocessing>=1.0.5 (from tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/28/6a/8c1f62c37212d9fc441a7e26736df51ce6f0e38455816445471f10da4f0a/Keras_Preprocessing-1.1.0-py2.py3-none-any.whl
Collecting tensorflow-estimator-2.0-preview (from tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/fc/25/2c861149cf7b97ed85b837a38822307d61a3e4418cf06b12c9187d462460/tensorflow_estimator_2.0_preview-1.14.0.dev2019060900-py2.py3-none-any.whl
Requirement already satisfied: setuptools>=41.0.0 in ./lib/python2.7/site-packages (from tb-nightly<1.15.0a0,>=1.14.0a0->tf-nightly-2.0-preview) (41.0.1)
Collecting futures>=3.1.1; python_version < "3" (from tb-nightly<1.15.0a0,>=1.14.0a0->tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/2d/99/b2c4e9d5a30f6471e410a146232b4118e697fa3ffc06d6a65efde84debd0/futures-3.2.0-py2-none-any.whl
Collecting werkzeug>=0.11.15 (from tb-nightly<1.15.0a0,>=1.14.0a0->tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/9f/57/92a497e38161ce40606c27a86759c6b92dd34fcdb33f64171ec559257c02/Werkzeug-0.15.4-py2.py3-none-any.whl
Collecting markdown>=2.6.8 (from tb-nightly<1.15.0a0,>=1.14.0a0->tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/c0/4e/fd492e91abdc2d2fcb70ef453064d980688762079397f779758e055f6575/Markdown-3.1.1-py2.py3-none-any.whl
Collecting funcsigs>=1; python_version < "3.3" (from mock>=2.0.0->tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/69/cb/f5be453359271714c01b9bd06126eaf2e368f1fddfff30818754b5ac2328/funcsigs-1.0.2-py2.py3-none-any.whl
Collecting h5py (from keras-applications>=1.0.6->tf-nightly-2.0-preview)
  Using cached https://files.pythonhosted.org/packages/53/08/27e4e9a369321862ffdce80ff1770553e9daec65d98befb2e14e7478b698/h5py-2.9.0-cp27-cp27mu-manylinux1_x86_64.whl
Installing collected packages: six, protobuf, enum34, absl-py, futures, werkzeug, markdown, grpcio, numpy, tb-nightly, backports.weakref, funcsigs, mock, termcolor, google-pasta, gast, h5py, keras-applications, wrapt, astor, keras-preprocessing, tensorflow-estimator-2.0-preview, tf-nightly-2.0-preview
Successfully installed absl-py-0.7.1 astor-0.8.0 backports.weakref-1.0.post1 enum34-1.1.6 funcsigs-1.0.2 futures-3.2.0 gast-0.2.2 google-pasta-0.1.7 grpcio-1.21.1 h5py-2.9.0 keras-applications-1.0.8 keras-preprocessing-1.1.0 markdown-3.1.1 mock-3.0.5 numpy-1.16.4 protobuf-3.8.0 six-1.12.0 tb-nightly-1.14.0a20190610 tensorflow-estimator-2.0-preview-1.14.0.dev2019060900 termcolor-1.1.0 tf-nightly-2.0-preview-2.0.0.dev20190608 werkzeug-0.15.4 wrapt-1.11.1
(tf2.0) mihaimaruseac@ankh:/tmp/virtual_pip/tf2.0$ python
Python 2.7.16 (default, Apr  6 2019, 01:42:57) 
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
>>> tf.summary
<module 'tensorboard.summary._tf.summary' from '/tmp/virtual_pip/tf2.0/local/lib/python2.7/site-packages/tensorboard/summary/_tf/summary/__init__.pyc'>
>>> 
(tf2.0) mihaimaruseac@ankh:/tmp/virtual_pip/tf2.0$ python -c "import tensorflow as tf; tf.summary"; echo $?
0
```

**Comment by mihaimaruseac** [comment]
Got it, was able to reproduce the exact example you gave

```bash
(tf2.0) mihaimaruseac@ankh:/tmp/virtual_pip/tf2.0$ python -c "import tensorflow.summary"; echo $?
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/buildtools/current/sitecustomize/sitecustomize.py", line 152, in SetupPathsAndImport
    return real_import(name, globals, locals, fromlist, level)
ImportError: No module named summary
1
```

**Comment by wchargin** [comment]
@mihaimaruseac: Right. Those discrepancies are exactly why we test so
many import permutations in our smoke test:

https://github.com/tensorflow/tensorboard/blob/7a4ef58a4e8e6a9db24b7b72ac2102e832dd590b/tensorboard/pip_package/build_pip_package.sh#L181-L193

Currently, `import_attr` and `import_from` work, but `import_as` fails.

**Comment by mihaimaruseac** [comment]
I have a fix, testing against tf1.* and tf.2* and on both pythons to make sure it works properly.

**Comment by mihaimaruseac** [comment]
Fix is in 2f0e7e39109ff7f40021425494f3c9d84a29b59c, sorry it took so long.

**Comment by wchargin** [comment]
Fixed in `tf-nightly-2.0-preview==2.0.0.dev20190613`. Thank you!
